.github/workflows/wheel_tests_and_release.yml

```diff
@@ -48,6 +45,14 @@ jobs:
           python -m build --no-isolation --skip-dependency-check --sdist .
           ls -la ${{ github.workspace }}/dist
 
+      - name: Verify the distribution
+        run: pipx run twine check --strict dist/*
```

Comment:
Beyond this workflow, the only area `twine` is used is

https://github.com/scikit-image/scikit-image/blob/765e568b9ff694f4680e4dbd76888ccdce664900/tools/upload_wheels.sh#L3-L5

and `tools/upload_wheels.sh` seems to no longer be used anywhere in this repository

```console
$ git grep "twine"
.github/workflows/wheel_tests_and_release.yml:        run: pipx run twine check --strict dist/*
tools/upload_wheels.sh:if [ "`which twine`" == "" ]; then
tools/upload_wheels.sh:    echo "twine not on path; need to pip install twine?"
$ git grep "upload_sheels
```

so instead of installing `twine` into the global environment can just use the pre-installed instance of `pipx` on the runner to get it (in a virtual environment) for a distribution metadata check.

Reply:
Should we not pin the version of twine, to avoid the threat of a hacked version of twine modifying uploaded wheel?

Reply:
This version of `twine` doesn't do anything with the upload. All of that is handled through the `pypa/gh-action-pypi-publish` GitHub action below.

Reply:
If you are also concerned with `twine` getting compromised and then modifying the wheel during the `twine check` command then you could pin to a trusted version, though I don't necessarily know if that's needed. If you were concerned you could also do a checksum before and after `twine check` ran to verify that it hasn't changed the distributions at all.

Reply:
I'm probably being paranoid; we install a lot of tools that can be compromised. Somehow, in my head, that one seemed like a good target because it does uploading. Feel free to do the checksumming, but not necessary.

Reply:
You're not wrong, so we could do this: https://github.com/scikit-image/scikit-image/pull/7427#discussion_r1612170116

(Posting here as I messed up the view on the review tab)